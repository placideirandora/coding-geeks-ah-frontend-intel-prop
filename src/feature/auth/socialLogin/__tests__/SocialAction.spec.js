import configureMockStore from 'redux-mock-store';
import dotenv from 'dotenv';
import thunk from 'redux-thunk';
import sinon from 'sinon';
import {
  social, authFail, authUser
} from '../SocialAction';
import * as actionTypes from '../SocialActionTypes';

dotenv.config();
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const user = {
  username: 'Eric6',
  role: 'user',
  email: 'eric.malaba6@gmail.com',
  exp: 1571291562,
  iat: 1571205162,
  id: 2
};

describe('Should create an action', () => {
  beforeAll(() => {
    sinon.stub(window.location, 'replace');
  });
  it('Should test LOGIN_SUCCESS type', () => {
    const source = '?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MiwiZW1haWwiOiJlcmljLm1hbGFiYTZAZ21haWwuY29tIiwicm9sZSI6InVzZXIiLCJ1c2VybmFtZSI6IkVyaWM2IiwiaWF0IjoxNTcxMjA1MTYyLCJleHAiOjE1NzEyOTE1NjJ9.ZdMucW9PC4K9PLJKX1G7cnWyHA2ihGfcBqhRT1-q4Co#';
    delete window.location;
    window.location = { search: source };
    const action = [{
      type: actionTypes.LOGIN_SUCCESS,
      user,
    }];
    const store = mockStore({});
    const url = window.location.search;
    store.dispatch(authUser(url));
    expect(store.getActions()).toEqual(action);
    expect(store.getActions()).toMatchSnapshot();
  });
  it('Should test LOGIN_FAIL type', () => {
    const store = mockStore({});
    store.dispatch(authFail('something went wrong'));
    expect(store.getActions()).toMatchSnapshot();
  });
  it('Should return error upon invalid login', () => {
    const source = '?token=';
    delete window.location;
    window.location = { search: source };
    const store = mockStore({});
    store.dispatch(authUser(source));
    expect(store.getActions()).toMatchSnapshot();
  });
  it('Should return error upon invalid login', () => {
    const source = 1;
    delete window.location;
    window.location = { search: source };
    const store = mockStore({});
    store.dispatch(social(source));
    expect(store.getActions()[0].type).toEqual('LOGIN_FAIL');
  });
});
