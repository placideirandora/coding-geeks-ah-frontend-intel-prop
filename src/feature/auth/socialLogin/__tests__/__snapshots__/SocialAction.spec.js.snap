// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should create an action Should return error upon invalid login 1`] = `
Array [
  Object {
    "error": [InvalidTokenError: Invalid token specified: Cannot read property 'replace' of undefined],
    "type": "LOGIN_FAIL",
  },
]
`;

exports[`Should create an action Should test LOGIN_FAIL type 1`] = `
Array [
  Object {
    "error": "something went wrong",
    "type": "LOGIN_FAIL",
  },
]
`;

exports[`Should create an action Should test LOGIN_SUCCESS type 1`] = `
Array [
  Object {
    "type": "LOGIN_SUCCESS",
    "user": Object {
      "email": "eric.malaba6@gmail.com",
      "exp": 1571291562,
      "iat": 1571205162,
      "id": 2,
      "role": "user",
      "username": "Eric6",
    },
  },
]
`;
